packages/calypso-e2e/src/hooks.ts

```diff
@@ -64,24 +64,26 @@ export const setupHooks = ( callback: ( { page }: { page: Page } ) => void ): vo
 			await page.screenshot( { path: fileName } );
 		}
 		// Close the browser. This needs to be called before trying to access
-		// the video recording
+		// the video recording.
 		await page.close();
-		// Save video
+
 		if ( __CURRENT_TEST_FAILED__ ) {
-			const original = await ( page.video() as Video ).path();
+			// const original = await ( page.video() as Video ).path();
 			const destination = path.join(
 				artifactPath,
 				'screenshots',
 				`${ getTestNameWithTime( testName ) }.webm`
 			);
 			try {
-				await rename( original, destination );
+				// Save the failing test case with a specific name.
+				await page.video()?.saveAs( destination );
 			} catch ( err ) {
-				console.error( 'Failed to rename video of failing test case.' );
+				console.error( 'Failed to save video of failing test case.' );
```

Comment:
While you are in this area, I think we can improve this and re-throw the error (or at the very least, log it) to avoid missing information

Reply:
Asked a question with the maintainers on whether all methods in the Video class have guarantees: https://github.com/microsoft/playwright/issues/9324

Reply:
As discussed below with @dpasque, I will merge this change for now as currently failed runs have no video recorded.

The intention is to look at refactoring how the video and screenshot capture happens, taking into account the possibility of multiple contexts. Filed https://github.com/Automattic/wp-calypso/issues/56805 for this.